client/components/data/data.service.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+var app = angular.module('app');
+
+app.service('Data', function() {
+  var _students = [];
+
+  this.students = function() {
+    return _students;
+  };
+  this.setStudents = function(students) {
+    _students.length = 0;
```

Comment:
Setting the length of an array to 0 is equivalent of removing each element of the array. This could also be done by using `_students.splice(0)`.